llm/cpp/llm.cpp

```diff
@@ -50,37 +33,28 @@ int main(int argc, char* argv[]) try {
     ov::InferRequest detokenizer = core.compile_model(argv[3], "CPU").create_infer_request();
     std::shared_ptr<ov::Model> model = core.read_model(argv[1]);
     constexpr size_t BATCH_SIZE = 1;
-    std::map<std::string, ov::PartialShape> shapes = {
-        {"input_ids", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+    std::map<size_t, ov::PartialShape> shapes = {
+        {0, ov::PartialShape{
+            BATCH_SIZE, -1
         }},
-        {"attention_mask", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+        {1, ov::PartialShape{
+            BATCH_SIZE, -1
         }}
     };
-    for (const ov::Output<ov::Node>& input : model->inputs()) {
-        for (const std::string& name : input.get_names()) {
-            if (name.rfind("past_key_values", 0) == 0) {
-                ov::PartialShape shape = input.get_partial_shape();
-                shape[0] = BATCH_SIZE;
-                shapes.emplace(name, shape);
-                break;
-            }
-        }
+    std::vector<ov::Output<ov::Node>> inputs = model->inputs();
+    for (size_t idx = 2; idx < inputs.size(); ++idx) {
+        ov::PartialShape shape = inputs.at(idx).get_partial_shape();
+        shape[0] = BATCH_SIZE;
+        shapes.emplace(idx, shape);
     }
     model->reshape(shapes);
     ov::preprocess::PrePostProcessor p3(model);
```

Comment:
Why is this needed? What is original type in IR?

Reply:
ov::element::i64

Reply:
Can we explicitly override the input element type with `i32` in intel-optimum code export? In the code below we always override the type, and actually it will likely be the case for any generative model. We have knowledge about the model in the export code and can do it safely instead of just blindly overriding it here in the code.

From the other hand, if i64 is not avoidable as an input type for the model, then why tokenizer gives i32 and requires alignment in the user code? Both sides are under our (partial) control. Let's make them composable from the beginning.

Reply:
We have `model_connect` or similarly called function that can connect tokenizer and main model. Here we don't need the connection itself, but the connection supposes similar alignment on types. Can we provide a way to align types as a service from the tokenizer library, @apaniukov? But it is better to already produce tokenizer model aligned anyway.

Reply:
> Can we explicitly override the input element type with `i32` in intel-optimum export code? In the code below we always override the type, and actually it will likely be the case for any generative model. We have knowledge about the model in the export code and can do it safely instead of just blindly overriding it here in the code.
> 
> From the other hand, if i64 is not avoidable as an input type for the model, then why tokenizer gives i32 and requires alignment in the user code? Both sides are under our (partial) control. Let's make them composable from the beginning.
> 
> @eaidova, @AlexKoff88, could you provide your feedback.

i32 works for most of the models, AFAIK. But we cannot guarantee that it works for all of them. I would keep i64 in the HF code. But I don't mind if we use i32 in this specific sample which is for a limited set of models.

Reply:
Change the default tokenizer output type to i64. Output type can be declared in `convert_tokenizer` now.

Reply:
I think detokenizer input should be changed the same way

Reply:
@Wovchena yes, you are right. Pushed this change to the branch.

Reply:
Updated